habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -321,3 +328,1068 @@ def get_all_object_ids(sim: habitat_sim.Simulator) -> Dict[int, str]:
+    habitat_sim.physics.ManagedArticulatedObject,
+]:
+    """
+    Get a ManagedRigidObject or ManagedArticulatedObject from an object_id.
+
+    ArticulatedLink object_ids will return the ManagedArticulatedObject.
+    If you want link id, use ManagedArticulatedObject.link_object_ids[obj_id].
+
+    :param sim: The Simulator instance.
+    :param obj_id: object id for which ManagedObject is desired.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id.
+
+    :return: a ManagedObject or None
+    """
+
+    if ao_link_map is None:
+        # Note: better to pre-compute this and pass it around
+        ao_link_map = get_ao_link_id_map(sim)
+
+    rom = sim.get_rigid_object_manager()
+    if rom.get_library_has_id(obj_id):
+        return rom.get_object_by_id(obj_id)
+    aom = sim.get_articulated_object_manager()
+    if obj_id in ao_link_map:
+        return aom.get_object_by_id(ao_link_map[obj_id])
+
+    return None
+
+
+def get_obj_from_handle(
+    sim: habitat_sim.Simulator, obj_handle: str
+) -> Union[
+    habitat_sim.physics.ManagedRigidObject,
+    habitat_sim.physics.ManagedArticulatedObject,
+]:
+    """
+    Get a ManagedRigidObject or ManagedArticulatedObject from its handle.
+
+    :param sim: The Simulator instance.
+    :param obj_handle: object istance handle for which ManagedObject is desired.
+
+    :return: a ManagedObject or None
+    """
+
+    rom = sim.get_rigid_object_manager()
+    if rom.get_library_has_handle(obj_handle):
+        return rom.get_object_by_handle(obj_handle)
+    aom = sim.get_articulated_object_manager()
+    if aom.get_library_has_handle(obj_handle):
+        return aom.get_object_by_handle(obj_handle)
+
+    return None
+
+
+def get_object_set_from_id_set(
+    sim: habitat_sim.Simulator,
+    id_set: List[int],
+    ao_link_map: Optional[Dict[int, int]] = None,
+) -> List[
+    Union[
+        habitat_sim.physics.ManagedRigidObject,
+        habitat_sim.physics.ManagedArticulatedObject,
+    ]
+]:
+    """
+    Get the ManagedObjects from a set of object_ids.
+
+    :param sim: The Simulator instance.
+    :param id_set: The set of object ids for which ManagedObjects are desired.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id.
+
+    :return: a list of tuples, first element is a ManagedObject, second is an optional link index.
+    """
+
+    if ao_link_map is None:
+        # Note: better to pre-compute this and pass it around
+        ao_link_map = get_ao_link_id_map(sim)
+
+    rom = sim.get_rigid_object_manager()
+    aom = sim.get_articulated_object_manager()
+    rigids = [
+        (rom.get_object_by_id(ro_id), None)
+        for ro_id in id_set
+        if rom.get_library_has_id(ro_id)
+    ]
+    aos = [
+        (
+            aom.get_object_by_id(ao_link_map[ao_id]),
+            aom.get_object_by_id(ao_link_map[ao_id]).link_object_ids[ao_id],
```

Comment:
I called the `above` function and got a KeyError from this line (particularly `.link_object_ids[ao_id]`). This was thrown part way through evaluation in Hab-LLM, likely after a state change. 

I've been using a quick hotfix, but haven't dug deeper.

```python
aos = []
for ao_id in id_set:
    if ao_id in ao_link_map:
        aa = aom.get_object_by_id(ao_link_map[ao_id])
        if ao_id not in aom.get_object_by_id(ao_link_map[ao_id]).link_object_ids:
            continue
        bb = aom.get_object_by_id(ao_link_map[ao_id]).link_object_ids[ao_id]
        aos.append((aa,bb))
```